core/trino-main/src/main/java/io/trino/eventlistener/EventListenerManager.java

```diff
@@ -83,6 +86,7 @@ public void addEventListener(EventListener eventListener)
         requireNonNull(eventListener, "EventListener is null");
 
         providedEventListeners.add(eventListener);
+        lifeCycleManager.addInstance(eventListener);
```

Comment:
i think it's not unreasonable for `io.trino.spi.eventlistener.EventListener` to be informed about system shutdown

if we want that, this should be an explicit method on that interface though, not some implicit annotation-reading which is hard to discover by interface implementors, and may not even work across classloaders.

Reply:
Let's follow `io.trino.spi.connector.Connector#shutdown` practice

Reply:
Yes. It'd be better to dispatch the shutdown event from the `EventListenerManager`. It should have @PreDestroy method that will call all event listeners shutdown method

Reply:
This wouldn't work anyway, as the annotations in the plugin are in a different class loader than the engine.